homeassistant/components/enigma2/media_player.py

```diff
@@ -0,0 +1,221 @@
+    if discovery_info:
+        discovered_config = {
+            CONF_NAME: DEFAULT_NAME,
+            # Do not use the discovered port, as it is the streaming
+            # service port (8001) which is not what we want.
+            CONF_PORT: DEFAULT_PORT,
+            CONF_HOST: discovery_info['host'],
+            CONF_USERNAME: DEFAULT_USERNAME,
+            CONF_PASSWORD: DEFAULT_PASSWORD,
+            CONF_SSL: DEFAULT_SSL,
+            CONF_PREFER_PICON: DEFAULT_PREFER_PICON,
+        }
+        add_devices([Enigma2Device(discovery_info['hostname'],
+                                   discovered_config)], True)
+        return
+
+    add_devices([Enigma2Device(config[CONF_NAME], config)], True)
+
+
+class Enigma2Device(MediaPlayerDevice):
+    """Representation of an Enigma2 box."""
+
+    def __init__(self, name, config):
+        """Initialize the Enigma2 device."""
+        import openwebif.api
+        self._name = name
+        self.e2_box = \
+            openwebif.api.CreateDevice(host=config[CONF_HOST],
+                                       port=config[CONF_PORT],
+                                       username=config[CONF_USERNAME],
+                                       password=config[CONF_PASSWORD],
+                                       is_https=config[CONF_SSL],
+                                       prefer_picon=config[CONF_PREFER_PICON])
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        if self.e2_box.is_recording_playback:
+            return STATE_PLAYING
+        return STATE_OFF if self.e2_box.in_standby else STATE_ON
+
+    @property
+    def supported_features(self):
+        """Flag of media commands that are supported."""
+        return SUPPORTED_ENIGMA2
+
+    def turn_off(self):
+        """Turn off media player."""
+        self.e2_box.turn_off()
+
+    def turn_on(self):
+        """Turn the media player on."""
+        self.e2_box.turn_on()
+
+    @property
+    def media_title(self):
+        """Title of current playing media."""
+        return self.e2_box.current_service_channel_name
+
+    @property
+    def media_series_title(self):
+        """Return the title of current episode of TV show."""
+        return self.e2_box.current_programme_name
+
+    @property
+    def media_channel(self):
+        """Channel of current playing media."""
+        return self.e2_box.current_service_channel_name
+
+    @property
+    def media_content_id(self):
+        """Service Ref of current playing media."""
+        return self.e2_box.current_service_ref
+
+    @property
+    def media_content_type(self):
+        """Type of video currently playing."""
+        return MEDIA_TYPE_TVSHOW
+
+    @property
+    def is_volume_muted(self):
+        """Boolean if volume is currently muted."""
+        return self.e2_box.muted
```

Comment:
Are all properties I/O free?

Reply:
Yes. Only the `CreateDevice` and the `update()` methods do I/O now in the pypi module.

Reply:
CreateDevice should then be outside of `__init__` as it will otherwise block the initialization (or that's my understanding :-))

Reply:
Mmm ok I see what you mean. Is there any way I can do a once off I/O call when the device gets created? I need to do that once to read the channel list, but don’t want to do it on every update()